### PR TITLE
Rename vllm.logging to vllm.logging_utils

### LINKED ISSUE #10133: [help wanted]: rename vllm/logging module to avoid shadowing builtin logging module
### Anything you want to discuss about vllm.

see https://github.com/vllm-project/vllm/issues/2021#issuecomment-2463435586 for more details.

we'd better avoid having a module that can potentially shadow python's builtin module, so that error trace can be clear.

solution could be:

rename `vllm.logging` to `vllm.logging_utils`

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by flozi00**
Hopefully Fixes #10133

Done from mobile, untested at the moment

### CODE REVIEW

File: vllm/logging_utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from vllm.logging_utils.formatter import NewLineFormatter
+
+__all__ = [
+    "NewLineFormatter",
+]
```
Comment (russellb) [inline vllm/logging_utils/__init__.py:5]: It would be ideal if this file was a `git mv` as well, like `formatter.py`

**Review by russellb — CHANGES_REQUESTED** [review]
The commit message needs a `Signed-off-by` header to pass the DCO check.

As for the changes, they look correct, though I'm not sure this is worth fixing. Avoiding naming collisions is an impossible task if we want to avoid all possible collisions to allow running code from within any arbitrary vllm source directory.

In any case, that's just an opinion and it's not my call to make.

**Comment by russellb** [comment]
@youkaichao FYI, since you filed the corresponding issue

File: vllm/logging_utils/__init__.py (hunk 1 above)
Comment (youkaichao) [reply vllm/logging_utils/__init__.py:5]: the content changes from  `vllm.logging` to `vllm.logging_utils`

**Review by youkaichao — APPROVED** [review]
LGTM!

**Comment by youkaichao** [comment]
> Avoiding naming collisions is an impossible task if we want to avoid all possible collisions to allow running code from within any arbitrary vllm source directory.

we only do it in case-by-case, only do it when we find annoying cases.

File: vllm/logging_utils/__init__.py (hunk 1 above)
Comment (russellb) [reply vllm/logging_utils/__init__.py:5]: only 1 line changes, no

Comment (russellb) [reply vllm/logging_utils/__init__.py:5]: actually, it's correct in git, and just not displayed like that in github's diff viewer.

**Pull request merged**